### ISSUE TITLE
Could not assign string or regex value for `origin`

### ISSUE THREAD

**Issue by surtr1st** [elysiajs/elysia-cors#16]
Hi, how can I resolve this `CORS` error? When set as a string or regex, this will show an error that says the origin prop is an empty string.

```
Response to preflight request doesn't pass access control check: The 'Access-Control-Allow-Origin' header contains the invalid value ''. Have the server send the header with a valid value, or, if an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

But when I use other approaches like boolean values or customize logic with functions, it will cause:

```
No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

This is my current usage:
```typescript
new Elysia()
  .use(
    cors({
      origin: ({ headers }) => {
        const origin = headers.get("origin");
        return origin === ALLOWED_ORIGIN;
      },
      allowedHeaders: ["Content-Type", "Accept", "Authorization"],
      methods: ["GET", "POST", "PUT", "PATCH", "DELETE", "OPTIONS"],
    }),
  )
  .group("/api", (app) => app.use(auth).use(user))
  .listen(3000);
```

**Comment by Kiyo5hi** [comment]
Dup: #7

**Comment by dgpt** [comment]
+1 I'm also seeing this and cannot seem to find an adequate resolution :/

**Comment by cannap** [comment]
same 

{ origin: ({ headers }) => {
  const origin = headers.get('origin')
  return origin === 'https://xxxx'
}, 
This works

**Comment by SaltyAom** [comment]
Can you verify if this was fixed in the latest 1.0.4 patch?

**Comment by surtr1st** [comment]
> Can you verify if this was fixed in the latest 1.0.4 patch?

Sorry, this has been quite a while. I need to re-adjust the version and fix some old code in the new version of Elysia.

But yes, the latest version of the `cors` plugin has resolved the problem.

Previously, I'd found the issue within my code, yet I wasn't sure if I was doing it the wrong way or if just the plugin was bugged.

My issue at that time was the hook I'm injected with; if I remember correctly, it worked when I removed all `beforeHandle`. The version of Elysia used was around 0.7.x, and the plugin was also around that.

However, after I tried the latest version of this plugin (specifically 1.0.4), I didn't need to remove all those hooks anymore, and it worked perfectly.
